enrichments/trace/internal/elastic/span.go

```diff
@@ -361,6 +370,9 @@ func (s *spanEnrichmentContext) setSpanTypeSubtype(span ptrace.Span) {
 	case s.isHTTP:
 		spanType = "external"
 		spanSubtype = "http"
+	case s.isGenAi:
+		spanType = "external"
+		spanSubtype = "genai"
```

Comment:
Did you consider `spanType="genai"` and `spanSubtype=s.genaiSystem`?

Also possible debate on `genai` vs `gen_ai`.

Reply:
I'm not sure how the Kibana UI is using span.type vs span.subtype there.

Reply:
For db it shows the db system, which is the subtype:
<img width="443" alt="Screenshot 2024-12-03 at 16 49 30" src="https://github.com/user-attachments/assets/0af59c43-328d-4eec-87e7-d04a9b5cf51e">

So, I think here it aways shows the subtype. 

`gen_ai.system` has a very limited set of well known values: https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/gen-ai.md

But then SemConv says:
> If one of them applies, then the respective value MUST be used; otherwise, a custom value MAY be used.

That last part with custom value puts me off a bit.

But no strong opinion from my side either. I think both options are fine.

Reply:
I'll find out once testing this :)

Reply:
> That last part with custom value puts me off a bit.

`db.system` has the same language in semconv:

"`db.system` has the following list of well-known values. If one of them applies, then the respective value MUST be used; otherwise, a custom value MAY be used."


So I think using `gen_ai.system` would be a "good thing" (TM).

Reply:
Updated to use the gen_ai.system value as span subtype and "genai" as type